ci/docker/min.build.Dockerfile

```diff
@@ -20,5 +21,22 @@ RUN rm -rf python/ray/thirdparty_files
 # install test requirements
 RUN python -m pip install -U pytest==7.0.1 numpy
 
-# core pre-release denpendencies
-RUN ./ci/env/install-core-prerelease-dependencies.sh
+# core dependencies
+RUN if [[ "${EXTRA_DEPENDENCY}" == "core" ]]; then \
+  ./ci/env/install-core-prerelease-dependencies.sh; \
+fi
+
+# train dependencies
+RUN if [[ "${EXTRA_DEPENDENCY}" == "ml" ]]; then \
+  python -m pip install -U "ray[tune]"; \
+fi
+
+# default dependencies 
+RUN if [[ "${EXTRA_DEPENDENCY}" == "default" ]]; then \
+  python -m pip install -U "ray[default]"; \
+fi
+
+# serve dependencies 
+RUN if [[ "${EXTRA_DEPENDENCY}" == "serve" ]]; then \
+  python -m pip install -U "ray[serve]"; \
+fi
```

Comment:
should use `<<EOF` and multiline thing here.. this is adding many extra useless layers to the image.

Reply:
sound good